src/util/getUserRef.ts

```diff
@@ -0,0 +1,7 @@
+import { State } from './initialState'
+import { Ref } from '../types'
+
+/**
+ * Get the user ref from an authenticated user's details stored in the state.
+ */
+export const getUserRef = (state: State): Ref | null => window.firebase?.database().ref('users/' + state.user!.uid)
```

Comment:
You changed the return type but I think you forgot to change the actual function so it returns `null` if the user is not logged in.